specification/hugr.md

```diff
@@ -698,66 +719,76 @@ See [Type System](#type-system) for more on Resources.
 # Import other header files to use their custom types
 imports: [Quantum]
 
-# Declare custom types
-types:
-- name: QubitVector
-  # Opaque types can take type arguments, with specified names
-  args: [size]
-
-# Declare operations which aren't associated to a resource
-operations:
-- name: measure
-  description: "measure a qubit"
-  # We're going to implement this using ops defined in the "Quantum" resource
-  resource_reqs: [Quantum] 
```

Comment:
I move that this requirement is either (a) use of types defined in `LinAlg`, i.e. the `complex_matrix` type mentioned  [in this comment](https://github.com/CQCL-DEV/hugr/pull/101/files#r1221438999); or (b) a property of the implementation, i.e. the rust/binary code (for what sets of resources `try_lower` will succeed or fail), rather than the interface.

That is, at present the YAML does not specify when `try_lower` will succeed or fail.